src/ElmSharp/ElmSharp/InputGenerator.cs

```diff
@@ -0,0 +1,242 @@
+        /// </summary>
+        Update,
+
+        /// <summary>
+        /// Finger release. It is same a behavior release your finger on touch screen.
+        /// </summary>
+        End,
+
+        /// <summary>
+        /// Max.
+        /// </summary>
+        /// <remarks>
+        /// Deprecated since 3.0.
+        /// </remarks>
+        Max = 10,
+    }
+
+    /// <summary>
+    /// InputGenerator provides functions to initialize/deinitialize input devices and to generation touch / key events.
+    /// </summary>
+    /// <privilege>
+    /// http://tizen.org/privilege/inputgenerator
+    /// </privilege>
+    /// <remarks>
+    /// This is not for use by third-party applications.
+    /// </remarks>
+    /// <since_tizen> preview </since_tizen>
+    public class InputGenerator : IDisposable
+    {
+        IntPtr _handle = IntPtr.Zero;
+        bool _isDisposed = false;
+
+        /// <summary>
+        /// Creates and initializes a new instance of the InputGenerator class.
+        /// </summary>
+        /// <param name="deviceType">The device type want to generate events</param>
+        /// <since_tizen> preview </since_tizen>
+        public InputGenerator(InputDeviceType deviceType)
+        {
+            _handle = Interop.Eutil.efl_util_input_initialize_generator((int)deviceType);
+        }
+
+        /// <summary>
+        /// Creates and initializes a new instance of the InputGenerator class with given name.
+        /// </summary>
+        /// <param name="deviceType">The device type want to generate events</param>
+        /// <param name="name">The device name (maximum 31 characters)</param>
+        /// <since_tizen> preview </since_tizen>
+        public InputGenerator(InputDeviceType deviceType, string name)
+        {
+            _handle = Interop.Eutil.efl_util_input_initialize_generator_with_name((int)deviceType, name);
+        }
+
+        /// <summary>
+        /// Destroys the InputGenerator object.
+        /// </summary>
+        ~InputGenerator()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Destroys the current object.
+        /// </summary>
+        /// <since_tizen> preview </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases all the resources currently used by this instance.
+        /// </summary>
+        /// <param name="disposing">
+        /// true if the managed resources should be disposed,
+        /// otherwise false.
+        /// </param>
+        /// <since_tizen> preview </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_isDisposed)
+                return;
+
+            if (disposing)
+            {
+                Interop.Eutil.efl_util_input_deinitialize_generator(_handle);
+            }
+
```

Comment:
Usually unmanaged object always released ,whether disposing is true or not
Is there some issue when execute efl_util_input_deinitialize_generator API in GC thread?

Reply:
Right. This API don't have thread safety, so there is issue when execute in GC thread.
But resource and device will be managed on window system when process is killed.